net/socket/socket.h

```diff
@@ -94,18 +94,30 @@
 /* Macro to set socket errors */
 
 #ifdef CONFIG_NET_SOCKOPTS
+#  define _SO_CONN_SETERRNO(c,e) \
+    do \
+      { \
+        (c)->s_error = (int16_t)e; \
+        set_errno(e); \
+      } \
+    while (0)
+
 #  define _SO_SETERRNO(s,e) \
     do \
       { \
         if (s != NULL && (s)->s_conn != NULL) \
```

Comment:
let's move (s)->s_conn check to _SO_CONN_SETERRNO

Reply:
Done